audio/debug.c

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2009-2012 Chris McClelland
+// Copyright (C) 2017 Kyle Robbertze <krobbertze@gmail.com>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+/** \file debug.c
+ * Implements writing to serial for debugging purposes
+ */
+#include <fx2regs.h>
+#include "debug.h"
+
+#define PD3 0xB3
```

Comment:
This is only for fx2 mini devboard right?

Reply:
`PD3` is TMS on the Atlys board and not sure what it is used for on the Opsis?

Reply:
FX2 yes, I don't know of a pin on either the opsis or atlys I can pull for serial communication. Didn't manage to port serial over